### DISCUSSION TITLE
🚀 Feature Request: add validation to Wrangler to discourage underscores in Worker names

### DISCUSSION THREAD

**Discussion by petebacondarwin** [cloudflare/workers-sdk#13504] [Feature Requests]
### Describe the solution

It is not possible to create Workers in the dashboard that contain underscores.
This is good practice because many SSL providers treat URLs with underscores as invalid.
Underscores in Worker names can also break remote dev in Wrangler.

This can result in difficult to diagnose problems as discussed in this issue: https://github.com/cloudflare/workers-sdk/issues/4921#issuecomment-1977456295

**Comment by AbhijeetShk** [comment]
Any pre-defined steps to add validation? Like file location, code to add and all

**Comment by petebacondarwin** [comment]
After discussion in the team we don't plan to completely block underscores in the next major release as this could cause considerable disruption to current users who already have Workers with this format.

Instead we should document this (via https://github.com/cloudflare/cloudflare-docs/issues/13355) and add only a validation warning in Wrangler.

The place to add the warning in Wrangler is https://github.com/cloudflare/workers-sdk/blob/7a0da671d2045c1a38c33074cab10db985c8d80b/packages/wrangler/src/config/validation-helpers.ts#L230-L247

**Comment by petebacondarwin** [comment]
Cases where this could be problematic to force people to remove underscores from their worker names are:
- customers using "subdomain" for their production Worker (e.g. `worker_name.myaccount.workers.dev`)
- customers using service bindings to workers with underscores
